cli/cmd/policy.go

```diff
@@ -0,0 +1,202 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	policyCmdState = struct {
+		File string
+		Repo bool
+		URL  string
+	}{}
+
+	// policyCmd represents the policy parent command
+	policyCmd = &cobra.Command{
+		Hidden: true,
+		Use:    "policy",
+		Short:  "manage policies",
+		Long: `Manage policies.
+
+NOTE: This feature is not yet available!`,
+	}
+
+	// policyCreateCmd represents the policy create command
+	policyCreateCmd = &cobra.Command{
+		Use:   "create",
+		Short: "create a policy",
+		Long: `Create a policy.
+
+A policy is represented in JSON format.
+The following attributes are minimally required:
+{
+    "policy_id": "lacework-example-1",
+    "title": "My Policy",
+    "enabled": false,
+    "lql_id": "MyQuery",
+    "severity": "high",
+    "description": "My Policy Description",
+    "remediation": "My Policy Remediation"
+}`,
+		Args: cobra.NoArgs,
+		RunE: createPolicy,
+	}
+)
+
+func init() {
+	// add the lql command
+	rootCmd.AddCommand(policyCmd)
+
+	// add sub-commands to the lql command
+	policyCmd.AddCommand(policyCreateCmd)
+
+	// run specific flags
+	setPolicySourceFlags(policyCreateCmd)
+}
+
+func setPolicySourceFlags(cmds ...*cobra.Command) {
+	for _, cmd := range cmds {
+		if cmd == nil {
+			return
+		}
+		action := strings.Split(cmd.Use, " ")[0]
+
+		// file flag to specify a policy from disk
+		cmd.Flags().StringVarP(
+			&policyCmdState.File,
+			"file", "f", "",
+			fmt.Sprintf("path to a policy to %s", action),
+		)
+		/* repo flag to specify a policy from repo
+		cmd.Flags().BoolVarP(
+			&policyCmdState.Repo,
+			"repo", "r", false,
+			fmt.Sprintf("id of a policy to %s via active repo", action),
+		)*/
+		// url flag to specify a policy from url
+		cmd.Flags().StringVarP(
+			&policyCmdState.URL,
+			"url", "u", "",
+			fmt.Sprintf("url to a policy to %s", action),
+		)
+	}
+}
+
+// for commands that take a policy as input
+func inputPolicy(cmd *cobra.Command, args []string) (string, error) {
```

Comment:
if you really don't need the arguments, don't pass them:
```suggestion
func inputPolicy(cmd *cobra.Command) (string, error) {
```

Reply:
This is most likely copy/pasta issue from `inputQuery` which does use `args`.  Will fix.

Reply:
Fixed.